packages/web-react/src/components/Button/Button.tsx

```diff
@@ -1,47 +1,78 @@
-import React from 'react';
+import React, { MouseEvent } from 'react';
 import classNames from 'classnames';
+import { WithChildren } from '../../types/main';
 
 type Color = 'primary' | 'secondary' | 'tertiary';
 
 const getButtonColorClassname = (color: Color): string => `Button--${color}`;
 
-export interface ButtonProps {
+export interface ButtonProps extends WithChildren {
   /**
    * Emotion color variant.
    */
   color: Color;
-  /**
-   * Button label.
-   */
-  label: string;
   /**
    * Function to be called on button click.
    */
-  onClick: () => void;
+  onClick?: (event: MouseEvent) => void;
   /**
    * HTML `type` attribute.
    */
   type: 'button' | 'submit';
+  /**
+   * Block alignment modification
+   */
+  block: boolean;
+  /**
+   * Disabling button
+   */
+  disabled: boolean;
+  /**
+   * Aria label
+   */
+  ariaLabel?: string;
 }
 
 export const Button = ({
   color,
-  label,
   onClick,
   type,
-}: ButtonProps): JSX.Element => (
-  <button
-    className={classNames('Button', getButtonColorClassname(color))}
-    onClick={onClick}
-    type={type}
-  >
-    {label}
-  </button>
-);
+  block,
+  ariaLabel,
+  disabled,
+  ...restAttributes
+}: ButtonProps): JSX.Element => {
+  const handleClick = (event: MouseEvent) => {
+    if (disabled) {
+      event.preventDefault();
+
+      return;
+    }
+
+    if (typeof onClick === 'function') {
```

Comment:
Well, type is checked by design with Typescript. You should check an existence instead: `if(onClick)` or `if(onClick != null)`

Reply:
changed